tests/datasources/test_ndvi.py

```diff
@@ -0,0 +1,214 @@
+    """
+    download_mock = mocker.patch(
+        (
+            "aatoolbox.datasources.usgs."
+            "usgs_ndvi._UsgsNdvi._download_ndvi_dekad"
+        )
+    )
+
+    def _mock_download(variable: str = "smoothed"):
+        ndvi = mock_ndvi(variable=variable)
+        fp = ndvi.download()
+        download_calls = download_mock.call_count
+        return fp, download_calls
+
+    return _mock_download
+
+
+@pytest.fixture
+def gdf():
+    """Create geodataframe for testing."""
+    d = {
+        "name": ["area_a", "area_b"],
+        "geometry": [
+            Polygon([(0, 0), (0, 2), (2, 2), (2, 0)]),
+            Polygon([(2, 0), (2, 2), (3, 2), (3, 0)]),
+        ],
+    }
+    return gpd.GeoDataFrame(d)
+
+
+@pytest.fixture
+def da():
+    """
+    Create raster input with date coordinate.
+
+    Used as side effect later to create 3
+    consecutive raster values for processing.
+    """
+
+    def _da(year_dekad):
+        date = dekad_to_date(year_dekad)
+        da = (
+            xr.DataArray(
+                [[[1 + year_dekad[1], 2, 3], [4, 5 + year_dekad[1], 6]]],
+                dims=("date", "y", "x"),
+                coords={
+                    "date": [date],
+                    "year": year_dekad[0],
+                    "dekad": year_dekad[1],
+                    "y": [1.5, 0.5],
+                    "x": [0.5, 1.5, 2.5],
+                },
+            )
+            .assign_coords({"modified": 1})
+            .rio.write_crs("EPSG:4326")
+        )
+        return da
+
+    return _da
+
+
+def test_download_smoothed(mock_download, mock_aa_data_dir):
+    """Test download for all NDVI variables."""
+    fp, call_count = mock_download(variable="smoothed")
+
+    assert fp == (mock_aa_data_dir / f"public/raw/glb/{DATASOURCE_BASE_DIR}")
+    assert call_count == 3
+
+
+def test_download_pct_median(mock_download, mock_aa_data_dir):
+    """Test download for all NDVI variables."""
+    fp, call_count = mock_download(variable="pct_median")
+
+    assert fp == (mock_aa_data_dir / f"public/raw/glb/{DATASOURCE_BASE_DIR}")
+
+    assert call_count
+
+
+def test_download_anomaly(mock_download, mock_aa_data_dir):
+    """Test download for all NDVI variables."""
+    fp, call_count = mock_download(variable="anomaly")
+
+    assert fp == (mock_aa_data_dir / f"public/raw/glb/{DATASOURCE_BASE_DIR}")
+
+    assert call_count
+
+
+def test_download_difference(mock_download, mock_aa_data_dir):
+    """Test download for all NDVI variables."""
+    fp, call_count = mock_download(variable="difference")
+
+    assert fp == (mock_aa_data_dir / f"public/raw/glb/{DATASOURCE_BASE_DIR}")
+
+    assert call_count
+
+
+def test_process_and_load(
+    mocker, mock_ndvi, mock_aa_data_dir, mock_country_config, gdf, da
+):
+    """Test processing NDVI values."""
+    ndvi = mock_ndvi(variable="smoothed")
+
+    # TODO: now created `load_raw` to be able to mock but would like
+    #  to do it from xr.load_dataset directly
```

Comment:
You can mock it directly, the reason why it didn't work for Tinka in the IRI tests is because `xr.load_dataset` is also needed in the testing module itself, but ends up being mocked too

Reply:
Aha I even figured it out relatively quickly. Dream life.